### PR TITLE
Hm/hybridize

### PR DESCRIPTION
**Pull request by haydenmcp**
- Hybridized the component

### CODE REVIEW

**Comment by haydenmcp** [comment]
@admwx7 Here is the paper-time-picker PR

File: bower.json [hunk 1]
```diff
@@ -20,23 +20,49 @@
...
+      },
+      "resolutions": {
+        "polymer": "^1.9",
+        "webcomponentsjs": "^0.7.2"
+      }
+    }
   },
   "resolutions": {
-    "webcomponentsjs": "^0.7.2"
+    "polymer": "^2.0",
```
Comment (admwx7) [inline bower.json]: Forcing resolution on Polymer breaks 1.x installs

**Review by admwx7 — CHANGES_REQUESTED** [review]
Missing changes to all files:
* @apply(--mixin) to @apply --mixin

Missing changes for paper-clock-selector:
* try wrapping _populate in a debounce, Polymer2 doesn't wait for all properties to be defined before it will run an observer so this will be a large performance hit if we don't mitigate it somehow
* similar case goes for _zeroPadChanged, _numbers could potentially be undefined in Polymer2 so swap `_numbers.forEach` for `(_numbers || []).forEach`

Missing changes for dialog styles:
* The top portion of the file has a style extension that needs wrapped

Missing changes for paper-time-picker:
* The top of the file has a style extension that needs wrapped
* Short circuit _updateTime if hour / minute / enableSeconds is undefined (not falsey) or if enableSeconds is true and second is undefined - same reason about not all props getting set

**Comment by admwx7** [comment]
This fixes #53

File: demo/index.html [hunk 2]
```diff
@@ -40,7 +40,7 @@
   </dom-bind>
   <script>
     (function() {
-      var scope;
+      let scope;
```
Comment (admwx7) [inline demo/index.html]: This is a breaking change to an otherwise non-breaking PR

File: paper-clock-selector.html [hunk 3]
```diff
@@ -277,43 +284,45 @@
...
-          while($numbers.firstChild) {
-            $numbers.removeChild($numbers.firstChild);
-          }
+            // remove dom nodes since they'll be re-created
+            while($numbers.firstChild) {
+              $numbers.removeChild($numbers.firstChild);
+            }
 
-          var numbers = [];
+            const numbers = [];
```
Comment (admwx7) [inline paper-clock-selector.html]: Also breaking change

File: paper-clock-selector.html [hunk 4]
```diff
@@ -277,43 +284,45 @@
...
 
-          for (var i=0; i<this.count; i++) {
-            value = i;
-            display = null;
-            if (i === 0 && !this.useZero) {
-              value = this.count;
-            }
+            for (let i = 0; i<this.count; i++) {
```
Comment (admwx7) [inline paper-clock-selector.html]: Breaking

File: paper-time-picker-dialog-style.html [hunk 5]
```diff
@@ -1,39 +1,41 @@
...
-  };
-  --paper-time-picker-dialog-heading: {
-    margin-bottom: -56px;
-  };
-}
-</style>
+<custom-style>
+  <style is="custom-style">
+  /* mixin definitions */
+  :root {
```
Comment (admwx7) [inline paper-time-picker-dialog-style.html]: :root has been dropped

File: paper-time-picker.html [hunk 6]
```diff
@@ -375,6 +383,11 @@
           this.rawValue = rawValue;
         },
         _updateTime: function(hour, minute, second) {
+          if (hour === undefined || minute === undefined ||
```
Comment (admwx7) [inline paper-time-picker.html]: Also need to account for enableSeconds being undefined, which you may as well just pass into this function while you're at it : )

File: paper-time-picker-dialog-style.html (hunk 5 above)
Comment (admwx7) [reply paper-time-picker-dialog-style.html]: Replace this with `:host > *`

File: paper-time-picker.html [hunk 7]
```diff
@@ -374,9 +382,14 @@
           }
           this.rawValue = rawValue;
         },
-        _updateTime: function(hour, minute, second) {
+        _updateTime: function(hour, minute, second, enableSeconds) {
+          if (hour === undefined || minute === undefined ||
```
Comment (admwx7) [inline paper-time-picker.html:386]: add `|| enableSeconds === undefined` as well

**Review by admwx7 — CHANGES_REQUESTED** [review]
1 last change in paper-time-picker.htmlL386

File: paper-time-picker.html (hunk 7 above)
Comment (haydenmcp) [reply paper-time-picker.html:386]: If this was the most recent comment, I applied this check right below this statement :-)

**Review by admwx7 — APPROVED** [review]

**Pull request merged**